x-pack/test/fleet_api_integration/apis/space_awareness/agents.ts

```diff
@@ -150,13 +170,20 @@ export default function (providerContext: FtrProviderContext) {
     });
 
     describe('DELETE /agents/{id}', () => {
-      it('should allow to delete an agent in the same space', async () => {
+      it('should allow deleting an agent in the same space', async () => {
         const testSpaceDeleteAgent = await createFleetAgent(
           esClient,
           spaceTest1Policy2.item.id,
           TEST_SPACE_1
         );
         await apiClient.deleteAgent(testSpaceDeleteAgent, TEST_SPACE_1);
+        // TODO: apiClient.deleteAgent does not remove the document?
```

Comment:
This had this scratching my head 🤔
I've tried a lot of different combinations, but I've always seen 409 errors because of this test case when adding a `beforeEach` hook to other test cases that attempts to clean the Fleet agents index. From researching the failure, my guess is that the `deleteByQuery` called in the cleaning function fails because this temporary agent was already marked for deletion but the document still exists for some reason.
If I don't manually delete the document (lines below), then other tests will fail because this agent is not expected to exist.
I'd be happy to hear if I'm missing anything!

Reply:
`apiClient.deleteAgent` does not remove the document our delete call do not delete the document but mark the agent as inactive kind of a soft deletion.

Reply:
🤦‍♀️ That makes sense, thanks. I'll clean this a bit.